src/Calendar.jsx

```diff
@@ -152,9 +157,7 @@ let Calendar = React.createClass({
 
     //if the value changes reset views to the new one
     if ( !dates.eq(val, dateOrNull(this.props.value), VIEW_UNIT[view]))
-      this.setState({
-        currentDate: val ? new Date(val) : new Date()
-      })
+      this.changeCurrentDate(val)
```

Comment:
I am a bit concerned about this causing update thrash. Since the common case is that the Calendar will

- trigger a `value` change
- receive new `value`
- trigger a `currentDate` change
- receive new `currentDate`

It might make more sense to trigger a currentValue change at the same time as the value change [here](https://github.com/jquense/react-widgets/blob/master/src/Calendar.jsx#L296). The problem though is you still need some version of the above check in case a new value is set from outside. 

This might just be the option for now...I wonder if it makes sense to remove this behavior entirely though